pkg/daemon/ceph/osd/volume_test.go

```diff
@@ -260,6 +268,124 @@ var cephVolumeRAWTestResult = `{
 }
 `
 
+func TestInitializeBlockPVC(t *testing.T) {
+	executor := &exectest.MockExecutor{}
+	executor.MockExecuteCommandWithCombinedOutput = func(command string, args ...string) (string, error) {
+		logger.Infof("%s %v", command, args)
+		if args[1] == "ceph-volume" && args[2] == "raw" && args[3] == "prepare" && args[4] == "--bluestore" && args[7] == "--crush-device-class" {
+			return initializeBlockPVCTestResult, nil
+		} else if args[1] == "ceph-volume" && args[2] == "lvm" && args[3] == "prepare" && args[4] == "--bluestore" && args[7] == "--crush-device-class" {
+			return initializeBlockPVCTestResult, nil
+		}
+
+		return "", errors.Errorf("unknown command %s %s", command, args)
+	}
+
+	os.Setenv(oposd.CrushDeviceClassVarName, "foo")
+	defer os.Setenv(oposd.CrushDeviceClassVarName, "")
+	context := &clusterd.Context{Executor: executor}
+	clusterInfo := &cephconfig.ClusterInfo{
+		CephVersion: cephver.CephVersion{Major: 14, Minor: 2, Extra: 8},
```

Comment:
we need a test that when using `{Major: 14, Minor: 2, Extra: 7}` then `args[2] == "lvm"` and when `{Major: 14, Minor: 2, Extra: 8}` then `args[2] == "raw"`.

Reply:
changes done.

Reply:
Where? I don't see this being validated

Reply:
validating here [CephVersion](https://github.com/rook/rook/pull/5276/commits/cc103256437221aadf38d8df752c044fa424df4d#diff-4e603f5894e11f99481f4b301ba8e9a8R310).

Reply:
Sorry @subhamkrai (maybe it's me) but I'm reading my initial comment and I still don't see where you validate that:

* when ceph version is 14.2.7 then the lvm arg is used
* when ceph version is 14.2.8 or higher the raw arg is used

The `MockExecuteCommandWithCombinedOutput` has both conditions so you can fall into any of both and you still don't know if you have the correct flag.

Reply:
okay I see the point. Changes are done.